### PR TITLE
[PT] detach tensor in get_const_data

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Return detached tensor from `get_const_data` function

### Reason for changes

Memory leak

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
Reproducer

```python
import torch
results = []
for _  in range(100):
    a = torch.rand((1,3,1000, 1000),requires_grad=True)
    b = torch.rand((1,3,1000, 1000),requires_grad=True)
    r = torch.sum(a + b)
    results.append(r)
```

image

```python
import torch
results = []
for _  in range(100):
    a = torch.rand((1,3,1000, 1000),requires_grad=True)
    b = torch.rand((1,3,1000, 1000),requires_grad=True)
    r = torch.sum(a.detach() + b.detach())
    results.append(r)
```
image

**Review by alexsu52 — COMMENTED** [review]
Do you have idea how to check that you covered all places in NNCF code?

**Comment by AlexanderDokuchaev** [comment]
No, but problem actual only for case when original tensor is replaced to new in the model which was calculated by using original.

**Review by kshpv — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]
